File: CodingSamples/Java/Runtime/ReflectionTest1/Program1.java
class Program {

	private static void present(Customer info) {
		System.out.println("<Customer>");
		System.out.printf("  <id>%s</id>%n", info.id());
		System.out.printf("  <purchase>%s</purchase>%n", info.purchase());
		System.out.printf("  <rating>%s</rating>%n", info.rating());
		System.out.println("</Customer>");
		System.out.println();
	}

	private static void present(Item info) {
		System.out.println("<Item>");
		System.out.printf("  <name>%s</name>%n", info.name());
		System.out.printf("  <brand>%s</brand>%n", info.brand());
		System.out.println("</Item>");
		System.out.println();

	}

	public static void main(String[] args) {
		String uid = System.getProperty("user.name");
		present(new Customer(uid, 45000, 3));	
		present(new Item("cpu", "intel"));
	}
}

File: CodingSamples/DotNet/Runtime/ReflectionTest1/DemoApp/Program1.cs
class Program
{
    private static void Present(Item info)
    {
        Console.WriteLine("<Item>");
        Console.WriteLine("  <Name>{0}</Name>", info.Name);
        Console.WriteLine("  <Brand>{0}</Brand>", info.Brand);
        Console.WriteLine("</Item>");
        Console.WriteLine();
    }

    private static void Present(Customer info)
    {
        Console.WriteLine("<Customer>");
        Console.WriteLine("  <Id>{0}</Id>", info.Id);
        Console.WriteLine("  <Purchase>{0}</Purchase>", info.Purchase);
        Console.WriteLine("  <Rating>{0}</Rating>", info.Rating);
        Console.WriteLine("</Customer>");
        Console.WriteLine();
    }

    public static void Main(string[] args)
    {
        Present(new Item("cpu", "intel"));
        Present(new Customer("Jack", 23000, 4));
    }
}
